src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SupplierAssessmentService.kt

```diff
@@ -27,4 +32,29 @@ class SupplierAssessmentService(
     )
     return referralRepository.save(referral)
   }
+
+  fun updateSupplierAssessmentAppointment(
+    referral: Referral,
+    durationInMinutes: Int,
+    appointmentTime: OffsetDateTime,
+    createdByUser: AuthUser
+  ): SupplierAssessment {
+    if (referral.supplierAssessment!!.appointments.size == 0) {
+      referral.supplierAssessment!!.appointments.add(
+        appointmentRepository.save(
+          Appointment(
+            id = UUID.randomUUID(),
+            appointmentTime = appointmentTime,
+            durationInMinutes = durationInMinutes,
+            createdBy = authUserRepository.save(createdByUser),
+            createdAt = OffsetDateTime.now()
+          )
+        )
+      )
+    } else {
+      referral.supplierAssessment!!.appointment.durationInMinutes = durationInMinutes
```

Comment:
do we want the same endpoint for both scheduling and rescheduling appointments?

could we delegate this out to an appointments service that both this and the action plan session code can call?

e.g. `AppointmentsService.bookOrRescheduleAppointment(appointment: Appointment?, durationInMinutes: Int, appointmentTime: OffsetDateTime, actor: AuthUser)`